web/server/web_client.c

```diff
@@ -1303,7 +1373,16 @@ void web_client_process_request(struct web_client *w) {
                 return;
             }
             break;
-
+#ifdef ENABLE_HTTPS
+        case HTTP_VALIDATION_REDIRECT:
+        {
+            buffer_flush(w->response.data);
+            w->response.data->contenttype = CT_TEXT_HTML;
+            buffer_strcat(w->response.data, "<!DOCTYPE html><!-- SPDX-License-Identifier: GPL-3.0-or-later --><html><body onload=\"window.location ='https://'+ window.location.hostname + ':' + window.location.port +  window.location.pathname\">Redirecting to safety connection, case your browser does not support redirection, please click <a onclick=\"window.location.href ='https://'+ window.location.hostname + ':' + window.location.port +  window.location.pathname\">here</a>.</body></html>");
+            w->response.code = 301;
```

Comment:
Issued an http request and it did redirect, but it took a very long time (30sec, i.e. there was a timeout). I saw the following:
```
2019-05-22 11:23:44: netdata ERROR : WEB_SERVER[static1] : (1092@/home/chri:web_client_send): Cannot send HTTP headers to web client. (errno 32, Broken pipe)
2019-05-22 11:23:44: netdata ERROR : WEB_SERVER[static1] : (1133@/home/chri:web_client_send): HTTP headers failed to be sent (I sent 386 bytes but the system sent -1 bytes). Closing web client.
```
After that, I actually did get the dashboard.
It's really strange because the first couple of times I did it, it redirected fine. Then I repeated the test another 5 times and got a timeout each time for the first https request after the redirect. At the same time, going directly to the https URL responded immediately. 

On 27-May I did more tests. First time, I got the netdata error you get when something fails (alert couldn't download something). Second time, got the 30s timeout and then got the redirect. Third time, redirect worked.

Reply:
This problem has a direct relationship with the problem that Ilya reported early today, I already found the place in the code where we are having this problem, I am working to fix now. The problem is that I am doing the redirect, but I did not close the connection properly.